--- ./libmdb.cvs/catalog.c	2005-06-17 04:42:18.000000000 +0200
+++ ./libmdb/catalog.c	2007-11-09 22:05:36.265625000 +0100
@@ -13,8 +13,8 @@
  *
  * You should have received a copy of the GNU Library General Public
  * License along with this library; if not, write to the
- * Free Software Foundation, Inc., 59 Temple Place - Suite 330,
- * Boston, MA 02111-1307, USA.
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301, USA.
  */
 
 #include "mdbtools.h"
@@ -97,8 +97,8 @@
 	while (mdb_fetch_row(table)) {
 		type = atoi(obj_type);
 		if (objtype==MDB_ANY || type == objtype) {
-			// fprintf(stdout, "obj_id: %10ld objtype: %-3d obj_name: %s\n", 
-			// (atol(obj_id) & 0x00FFFFFF), type, obj_name); 
+			
+			
 			entry = (MdbCatalogEntry *) g_malloc0(sizeof(MdbCatalogEntry));
 			entry->mdb = mdb;
 			strcpy(entry->object_name, obj_name);
@@ -109,7 +109,7 @@
 			g_ptr_array_add(mdb->catalog, entry); 
 		}
 	}
-	//mdb_dump_catalog(mdb, MDB_TABLE);
+	
  
 	mdb_free_tabledef(table);
 
