--- ./libmdb.cvs/map.c	2005-02-11 17:24:21.000000000 +0100
+++ ./libmdb/map.c	2007-11-09 22:05:36.312500000 +0100
@@ -13,8 +13,8 @@
  *
  * You should have received a copy of the GNU Library General Public
  * License along with this library; if not, write to the
- * Free Software Foundation, Inc., 59 Temple Place - Suite 330,
- * Boston, MA 02111-1307, USA.
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301, USA.
  */
 
 #include "mdbtools.h"
@@ -114,7 +114,7 @@
 		pgnum = mdb_map_find_next(mdb, 
 				table->free_usage_map, 
 				table->freemap_sz, cur_pg);
-		//printf("looking at page %d\n", pgnum);
+		
 		if (!pgnum) {
 			/* allocate new page */
 			pgnum = mdb_alloc_page(table);
@@ -127,7 +127,7 @@
 		
 	} while (free_space < row_size);
 
-	//printf("page %d has %d bytes left\n", pgnum, free_space);
+	
 
 	return pgnum;
 }
