--- ./libmdb.cvs/money.c	2005-03-13 06:26:50.000000000 +0100
+++ ./libmdb/money.c	2007-11-09 22:05:36.656250000 +0100
@@ -13,8 +13,8 @@
  *
  * You should have received a copy of the GNU Library General Public
  * License along with this library; if not, write to the
- * Free Software Foundation, Inc., 59 Temple Place - Suite 330,
- * Boston, MA 02111-1307, USA.
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301, USA.
  */
 
 #include <stdio.h>
@@ -43,7 +43,7 @@
  */
 char *mdb_money_to_string(MdbHandle *mdb, int start)
 {
-	const int num_bytes = 8;
+	#define num_bytes 8
 	int i;
 	int neg=0;
 	unsigned char multiplier[MAXPRECISION], temp[MAXPRECISION];
