--- ../../_cxx_5.3.0/Src/cxxsupport.cxx	2004-11-14 21:52:52.000000000 +0100
+++ cxxsupport.cxx	2005-04-20 15:40:02.000000000 +0200
@@ -28,6 +28,11 @@ std::string Object::as_string() const
 	return static_cast<std::string>(str());
 	}
 
+List Object::dir () const
+        {
+        return List (PyObject_Dir (p), true);
+        }
+
 bool Object::isType (const Type& t) const
 	{ 
 	return type ().ptr() == t.ptr();
--- ../../_cxx_5.3.0/CXX/Objects.hxx	2004-11-14 21:52:52.000000000 +0100
+++ ../CXX/Objects.hxx	2005-04-20 15:40:04.000000000 +0200
@@ -27,6 +27,7 @@ namespace Py
 	class Type;
 	template<TEMPLATE_TYPENAME T> class SeqBase;
 	class String;
+        class List;
 	template<TEMPLATE_TYPENAME T> class MapBase;
 
 	// new_reference_to also overloaded below on Object
@@ -242,6 +243,8 @@ namespace Py
 
 		String repr () const; // the repr () representation
 
+                List dir () const; // the dir() list
+
 		bool hasAttr (const std::string& s) const
 			{
 			return PyObject_HasAttrString (p, const_cast<char*>(s.c_str())) ? true: false;
