Index: kpttask.cc
===================================================================
--- kpttask.cc	(revision 578211)
+++ kpttask.cc	(working copy)
@@ -1471,6 +1471,12 @@
         if (m_currentSchedule->startTime < m_currentSchedule->latestFinish) {
             f = m_currentSchedule->latestFinish - m_currentSchedule->startTime;
         }
+    } else if (m_effort->type() == Effort::Type_FixedDuration) {
+        if (m_currentSchedule->endTime.isValid()) {
+            if (m_currentSchedule->endTime < m_currentSchedule->latestFinish) {
+                f = m_currentSchedule->latestFinish - m_currentSchedule->endTime;
+            }
+        }
     } else {
         if (m_currentSchedule->workEndTime.isValid())
             if (m_currentSchedule->workEndTime < m_currentSchedule->latestFinish) {
