Index: kptrelation.cc
===================================================================
--- kptrelation.cc	(revision 556439)
+++ kptrelation.cc	(working copy)
@@ -74,6 +74,10 @@
     if (m_child == 0) {
         return false;
     }
+    if (m_child == m_parent) {
+        kdDebug()<<k_funcinfo<<"child == parent"<<endl;
+        return false;
+    }
     if (!m_parent->legalToLink(m_child))
         return false;
         
