Index: kptproject.h
===================================================================
--- kptproject.h	(revision 558898)
+++ kptproject.h	(working copy)
@@ -205,11 +205,9 @@
     virtual Calendar *findCalendar(const QString &id) const 
         { return id.isEmpty() ? 0 : calendarIdDict.find(id); }
     /// Remove the calendar with identity id from the register
-    virtual bool removeCalendarId(const QString &id) 
-        { return calendarIdDict.remove(id); }
+    virtual bool removeCalendarId(const QString &id);
     /// Insert the calendar with identity id
-    virtual void insertCalendarId(const QString &id, const Calendar *calendar)
-        { calendarIdDict.insert(id, calendar); }
+    virtual void insertCalendarId(const QString &id, const Calendar *calendar);
     
     /**
      * Setting a project to be baselined means the project data can not be edited anymore.
Index: kptcalendar.cc
===================================================================
--- kptcalendar.cc	(revision 558331)
+++ kptcalendar.cc	(working copy)
@@ -523,7 +523,8 @@
     delete m_weekdays; 
 }
 Calendar::Calendar(Calendar *calendar)
-    : m_days() {
+    : m_project(0),
+      m_days() {
     m_days.setAutoDelete(true);
     copy(*calendar);
 }
@@ -569,7 +570,7 @@
     }
     Calendar *c = findCalendar();
     if (c == this) {
-        //kdDebug()<<k_funcinfo<<"My id found, remove it"<<endl;
+        kdDebug()<<k_funcinfo<<"My id found, remove it"<<endl;
         removeId();
     } else if (c) {
         //can happen when making a copy
Index: kptproject.cc
===================================================================
--- kptproject.cc	(revision 558898)
+++ kptproject.cc	(working copy)
@@ -1068,6 +1068,16 @@
     return sch;
 }
 
+bool Project::removeCalendarId(const QString &id) {
+    kdDebug()<<k_funcinfo<<"id="<<id<<endl;
+    return calendarIdDict.remove(id); 
+}
+
+void Project::insertCalendarId(const QString &id, const Calendar *calendar) { 
+    kdDebug()<<k_funcinfo<<"id="<<id<<": "<<calendar->name()<<endl;
+    calendarIdDict.insert(id, calendar);
+}
+        
 #ifndef NDEBUG
 void Project::printDebug(bool children, QCString indent) {
 
Index: kptcalendarlistdialog.cc
===================================================================
--- kptcalendarlistdialog.cc	(revision 558331)
+++ kptcalendarlistdialog.cc	(working copy)
@@ -173,7 +173,7 @@
     QPtrListIterator<Calendar> it = list;
     for (; it.current(); ++it) {
         Calendar *c = new Calendar(it.current());
-        c->setProject(&p);
+        //c->setProject(&p);
         new CalendarListViewItem(*dia, dia->calendarList, c, it.current());
     }
     dia->setBaseCalendars();
Index: kptresource.cc
===================================================================
--- kptresource.cc	(revision 558331)
+++ kptresource.cc	(working copy)
@@ -322,8 +322,12 @@
 }
 
 Calendar *Resource::calendar(bool local) const {
-    if ( local == false && m_calendar == 0 && project() != 0)
+    if (!local && project() != 0 && (m_calendar == 0 || m_calendar->isDeleted())) {
         return project()->defaultCalendar();
+    }
+    if (m_calendar && m_calendar->isDeleted()) {
+        return 0;
+    }
     return m_calendar;
 }
 
@@ -366,7 +370,7 @@
     QDomElement me = element.ownerDocument().createElement("resource");
     element.appendChild(me);
 
-    if (m_calendar)
+    if (calendar(true))
         me.setAttribute("calendar-id", m_calendar->id());
     me.setAttribute("id", m_id);
     me.setAttribute("name", m_name);
