Index: kptproject.h
===================================================================
--- kptproject.h	(revision 528102)
+++ kptproject.h	(working copy)
@@ -1,6 +1,6 @@
 /* This file is part of the KDE project
    Copyright (C) 2001 Thomas Zander zander@kde.org
-   Copyright (C) 2004, 2005 Dag Andersen <danders@get2net.dk>
+   Copyright (C) 2004 - 2006 Dag Andersen <danders@get2net.dk>
 
    This library is free software; you can redistribute it and/or
    modify it under the terms of the GNU Library General Public
@@ -149,20 +149,20 @@
     /// Actual cost up to and including date
     virtual double actualCostTo(const QDate &date);
 
-    Calendar *defaultCalendar() { return m_defaultCalendar; }
+    Calendar *defaultCalendar() { return m_standardWorktime->calendar(); }
     QPtrList<Calendar> calendars();
     void addCalendar(Calendar *calendar);
     /// Returns the calendar with identity id.
     Calendar *calendar(const QString id) const;
 
     /**
-     * Defines the length of days, weeks, months and years.
+     * Defines the length of days, weeks, months and years
+     * and the standard working week.
      * Used for estimation and calculation of effort, 
      * and presentation in gantt chart.
      */    
     StandardWorktime *standardWorktime() { return m_standardWorktime; }
     void setStandardWorktime(StandardWorktime * worktime);
-    void setDefaultCalendar(Calendar *cal);
 
     /// Check if node par can be linked to node child.
     bool legalToLink(Node *par, Node *child);
@@ -233,7 +233,6 @@
     Accounts m_accounts;
     QPtrList<ResourceGroup> m_resourceGroups;
 
-    Calendar *m_defaultCalendar;
     QPtrList<Calendar> m_calendars;
 
     StandardWorktime *m_standardWorktime;
Index: kptcalendar.cc
===================================================================
--- kptcalendar.cc	(revision 528108)
+++ kptcalendar.cc	(working copy)
@@ -1,5 +1,5 @@
 /* This file is part of the KDE project
-   Copyright (C) 2003 - 2005 Dag Andersen <danders@get2net.dk>
+   Copyright (C) 2003 - 2006 Dag Andersen <danders@get2net.dk>
 
    This library is free software; you can redistribute it and/or
    modify it under the terms of the GNU Library General Public
@@ -929,6 +929,7 @@
         m_month = worktime->durationMonth();
         m_week = worktime->durationWeek();
         m_day = worktime->durationDay();
+        m_calendar = new Calendar(*(worktime->calendar()));
     } else {
         init();
     }
@@ -939,11 +940,20 @@
 }
 
 void StandardWorktime::init() {
-    // Some temporary sane default values
+    // Some sane default values
     m_year = Duration(0, 1760, 0);
     m_month = Duration(0, 176, 0);
     m_week = Duration(0, 40, 0);
     m_day = Duration(0, 8, 0);
+    m_calendar = new Calendar;
+    m_calendar->setName(i18n("Base"));
+    QPair<QTime, QTime> t = QPair<QTime, QTime>(QTime(8,0,0), QTime(16,0,0));
+    for (int i=0; i < 5; ++i) {
+        m_calendar->weekday(i)->addInterval(t);
+        m_calendar->weekday(i)->setState(Map::Working);
+    }
+    m_calendar->weekday(5)->setState(Map::NonWorking);
+    m_calendar->weekday(6)->setState(Map::NonWorking);
 }
 
 bool StandardWorktime::load(QDomElement &element) {
@@ -952,6 +962,18 @@
     m_month = Duration::fromString(element.attribute("month"), Duration::Format_Hour); 
     m_week = Duration::fromString(element.attribute("week"), Duration::Format_Hour); 
     m_day = Duration::fromString(element.attribute("day"), Duration::Format_Hour); 
+    
+    QDomNodeList list = element.childNodes();
+    for (unsigned int i=0; i<list.count(); ++i) {
+        if (list.item(i).isElement()) {
+            QDomElement e = list.item(i).toElement();
+            if (e.tagName() == "calendar") {
+                delete m_calendar;
+                m_calendar = new Calendar;
+                m_calendar->load(e);
+            }
+        }
+    }
     return true;
 }
 
@@ -963,6 +985,8 @@
     me.setAttribute("month", m_month.toString(Duration::Format_Hour));
     me.setAttribute("week", m_week.toString(Duration::Format_Hour));
     me.setAttribute("day", m_day.toString(Duration::Format_Hour));
+    
+    m_calendar->save(me);
 }
 
 #ifndef NDEBUG
Index: kptstandardworktimedialog.cc
===================================================================
--- kptstandardworktimedialog.cc	(revision 528102)
+++ kptstandardworktimedialog.cc	(working copy)
@@ -1,5 +1,5 @@
 /* This file is part of the KDE project
-   Copyright (C) 2004 Dag Andersen <danders@get2net.dk>
+   Copyright (C) 2004 - 2006 Dag Andersen <danders@get2net.dk>
 
    This library is free software; you can redistribute it and/or
    modify it under the terms of the GNU Library General Public
@@ -18,26 +18,79 @@
 */
 
 #include "kptstandardworktimedialog.h"
+#include "kptduration.h"
 #include "kptproject.h"
 #include "kptcalendar.h"
 #include "kptcommand.h"
 #include "kptintervaledit.h"
 #include "kptpart.h"
 
+#include <qgroupbox.h>
+#include <qheader.h>
 #include <QPushButton>
 #include <QSpinBox>
 #include <QComboBox>
 #include <QLabel>
+#include <QLayout>
 #include <QLineEdit>
 #include <qdatetimeedit.h>
 #include <qdatetime.h>
 
 #include <kdebug.h>
+#include <kcombobox.h>
+#include <kcalendarsystem.h>
+#include <kglobal.h>
 #include <klocale.h>
 #include <knuminput.h>
+#include <k3listview.h>
 
 namespace KPlato
 {
+class WeekdayListItem : public K3ListViewItem
+{
+public:
+    WeekdayListItem(Calendar *cal, int wd, K3ListView *parent, QString name, K3ListViewItem *after)
+    : K3ListViewItem(parent, after),
+      original(cal->weekday(wd)),
+      calendar(cal),
+      weekday(wd)
+    {
+        setText(0, name);
+        day = new CalendarDay(original);
+        if (day->state() == Map::NonWorking) {
+            setHours();
+        } else {
+            setText(1, KGlobal::locale()->formatNumber(day->duration().toDouble(Duration::Unit_h)));
+        }
+    }
+    ~WeekdayListItem() {
+        delete day;
+    }
+    void setHours() {
+        setText(1, "-");
+        day->clearIntervals();
+    }
+    void setIntervals(QPtrList<QPair<QTime, QTime> > intervals) {
+        day->setIntervals(intervals);
+        setText(1, KGlobal::locale()->formatNumber(day->duration().toDouble(Duration::Unit_h)));
+    }
+    void setState(int st) {
+        day->setState(st+1);
+    }
+    
+    KCommand *save(Part *part) {
+        KCommand *cmd=0;
+        if (*original != *day) {
+            cmd = new CalendarModifyWeekdayCmd(part, calendar, weekday, day);
+            day = 0;
+        }
+        return cmd;
+    }
+    CalendarDay *day;
+    CalendarDay *original;
+    Calendar *calendar;
+    int weekday;
+};
 
 StandardWorktimeDialog::StandardWorktimeDialog(Project &p, QWidget *parent, const char *name)
     : KDialogBase( Swallow, i18n("Standard Worktime"), Ok|Cancel, Ok, parent, name, true, true),
@@ -55,7 +108,7 @@
 }
 
 KMacroCommand *StandardWorktimeDialog::buildCommand(Part *part) {
-    kDebug()<<k_funcinfo<<endl;
+    //kDebug()<<k_funcinfo<<endl;
     QString n = i18n("Modify Standard Worktime");
     KMacroCommand *cmd = 0;
     if (m_original->year() != dia->inYear()) {
@@ -74,6 +127,14 @@
         if (cmd == 0) cmd = new KMacroCommand(n);
         cmd->addCommand(new ModifyStandardWorktimeDayCmd(part, m_original, m_original->day(), dia->inDay()));
     }
+    QListViewItem *item = dia->weekdayList->firstChild();
+    for (; item; item = item->nextSibling()) {
+        KCommand *c = static_cast<WeekdayListItem*>(item)->save(part);
+        if (c) {
+            if (cmd == 0) cmd = new KMacroCommand(n);
+            cmd->addCommand(c);
+        }
+    }
     return cmd;
     
 }
@@ -89,6 +150,10 @@
     if (!std) {
         m_std = new StandardWorktime();
     }
+    QBoxLayout *l = new QVBoxLayout(intervalBox);
+    m_intervalEdit = new IntervalEdit(intervalBox);
+    l->addWidget(m_intervalEdit);
+    
     m_year = m_std->year();
     m_month = m_std->month();
     m_week = m_std->week();
@@ -99,14 +164,42 @@
     week->setValue(m_week);
     day->setValue(m_day);
 
+    weekdayList->setSorting(-1);
+    weekdayList->header()->setStretchEnabled(true);
+    const KCalendarSystem * cs = KGlobal::locale()->calendar();
+    Calendar *cal = m_std->calendar();
+    if (cal) {
+        WeekdayListItem *item = 0;
+        for (int i = 0; i < 7; ++i) {
+            CalendarDay *day = cal->weekday(i);
+            if (day == 0) {
+                continue;
+            }
+            item = new WeekdayListItem(cal, i, weekdayList, cs->weekDayName(i+1), item);
+            weekdayList->insertItem(item);
+        }
+    }
     connect(year, SIGNAL(valueChanged(double)), SLOT(slotYearChanged(double)));
     connect(month, SIGNAL(valueChanged(double)), SLOT(slotMonthChanged(double)));
     connect(week, SIGNAL(valueChanged(double)), SLOT(slotWeekChanged(double)));
     connect(day, SIGNAL(valueChanged(double)), SLOT(slotDayChanged(double)));
     
+    connect(m_intervalEdit, SIGNAL(changed()), SLOT(slotIntervalChanged()));
+    connect(bApply, SIGNAL(clicked()), SLOT(slotApplyClicked()));
+    connect(weekdayList, SIGNAL(selectionChanged()), SLOT(slotWeekdaySelected()));
+    connect(state, SIGNAL(activated(int)), SLOT(slotStateChanged(int)));
+    
+    if (weekdayList->firstChild()) {
+        weekdayList->setSelected(weekdayList->firstChild(), true);
+        weekdayList->setCurrentItem(weekdayList->firstChild());
+    }
 }
 
 
+void StandardWorktimeDialogImpl::slotEnableButtonApply(bool on) {
+    bApply->setEnabled(on);
+}
+
 void StandardWorktimeDialogImpl::slotEnableButtonOk(bool on) {
     emit enableButtonOk(on);
 }
@@ -148,7 +241,52 @@
     slotEnableButtonOk(true);
 }
 
+void StandardWorktimeDialogImpl::slotIntervalChanged() {
+    //kDebug()<<k_funcinfo<<endl;
+    slotEnableButtonApply(true);
+}
 
+void StandardWorktimeDialogImpl::slotApplyClicked() {
+    //kDebug()<<k_funcinfo<<"state="<<state->currentItem()<<endl;
+    QListViewItem *item = weekdayList->firstChild();
+    for (; item; item = item->nextSibling()) {
+        if (item->isSelected()) {
+            //kDebug()<<k_funcinfo<<item->text(0)<<" selected"<<endl;
+            WeekdayListItem *wd = static_cast<WeekdayListItem*>(item);
+            wd->setState(state->currentItem());
+            if (state->currentItem() == 0) {
+                wd->setHours();
+            } else {
+                wd->setIntervals(m_intervalEdit->intervals());
+            }
+            slotEnableButtonOk(true);
+        }
+    }
+}
+
+void StandardWorktimeDialogImpl::slotWeekdaySelected() {
+    //kDebug()<<k_funcinfo<<"state="<<state->currentItem()<<endl;
+    
+    QListViewItem *item = weekdayList->firstChild();
+    for (; item; item = item->nextSibling()) {
+        if (item->isSelected()) {
+            //kDebug()<<k_funcinfo<<item->text(0)<<" selected"<<endl;
+            WeekdayListItem *wd = static_cast<WeekdayListItem*>(item);
+            state->setCurrentItem(wd->day->state()-1);
+            m_intervalEdit->setIntervals(wd->day->workingIntervals());
+            slotStateChanged(state->currentItem());
+            break;
+        }
+    }
+    editBox->setEnabled(item != 0);
+}
+
+void StandardWorktimeDialogImpl::slotStateChanged(int st) {
+    //kDebug()<<k_funcinfo<<"state="<<state->currentItem()<<endl;
+    intervalBox->setEnabled(st == 1); //Working
+    slotEnableButtonApply(st == 0);
+}
+
 }  //KPlato namespace
 
 #include "kptstandardworktimedialog.moc"
Index: standardworktimedialogbase.ui
===================================================================
--- standardworktimedialogbase.ui	(revision 528102)
+++ standardworktimedialogbase.ui	(working copy)
@@ -9,8 +9,8 @@
         <rect>
             <x>0</x>
             <y>0</y>
-            <width>228</width>
-            <height>112</height>
+            <width>417</width>
+            <height>368</height>
         </rect>
     </property>
     <property name="caption">
@@ -19,134 +19,272 @@
     <property name="whatsThis" stdset="0">
         <string>These values are used when you estimate the effort needed to complete a task.</string>
     </property>
-    <grid>
+    <vbox>
         <property name="name">
             <cstring>unnamed</cstring>
         </property>
         <property name="margin">
             <number>0</number>
         </property>
-        <widget class="QLabel" row="0" column="0">
+        <widget class="QLayoutWidget">
             <property name="name">
-                <cstring>textLabel1</cstring>
+                <cstring>layout10</cstring>
             </property>
-            <property name="text">
-                <string>Hours per year:</string>
-            </property>
-            <property name="buddy" stdset="0">
-                <cstring>year</cstring>
-            </property>
+            <grid>
+                <property name="name">
+                    <cstring>unnamed</cstring>
+                </property>
+                <widget class="QLabel" row="3" column="0">
+                    <property name="name">
+                        <cstring>textLabel4</cstring>
+                    </property>
+                    <property name="text">
+                        <string>Hours per day:</string>
+                    </property>
+                    <property name="buddy" stdset="0">
+                        <cstring>day</cstring>
+                    </property>
+                </widget>
+                <widget class="QLabel" row="1" column="0">
+                    <property name="name">
+                        <cstring>textLabel2</cstring>
+                    </property>
+                    <property name="text">
+                        <string>Hours per month:</string>
+                    </property>
+                    <property name="buddy" stdset="0">
+                        <cstring>month</cstring>
+                    </property>
+                </widget>
+                <widget class="QLabel" row="2" column="0">
+                    <property name="name">
+                        <cstring>textLabe3</cstring>
+                    </property>
+                    <property name="text">
+                        <string>Hours per week:</string>
+                    </property>
+                    <property name="buddy" stdset="0">
+                        <cstring>week</cstring>
+                    </property>
+                </widget>
+                <widget class="KDoubleSpinBox" row="1" column="1">
+                    <property name="name">
+                        <cstring>month</cstring>
+                    </property>
+                    <property name="focusPolicy">
+                        <enum>WheelFocus</enum>
+                    </property>
+                    <property name="maxValue">
+                        <number>744</number>
+                    </property>
+                    <property name="minValue">
+                        <number>1</number>
+                    </property>
+                    <property name="toolTip" stdset="0">
+                        <string>Number of working hours in a normal month.</string>
+                    </property>
+                </widget>
+                <widget class="KDoubleSpinBox" row="3" column="1">
+                    <property name="name">
+                        <cstring>day</cstring>
+                    </property>
+                    <property name="focusPolicy">
+                        <enum>WheelFocus</enum>
+                    </property>
+                    <property name="maxValue">
+                        <number>24</number>
+                    </property>
+                    <property name="minValue">
+                        <number>1</number>
+                    </property>
+                    <property name="toolTip" stdset="0">
+                        <string>Number of working hours in a normal day.</string>
+                    </property>
+                </widget>
+                <widget class="KDoubleSpinBox" row="0" column="1">
+                    <property name="name">
+                        <cstring>year</cstring>
+                    </property>
+                    <property name="focusPolicy">
+                        <enum>WheelFocus</enum>
+                    </property>
+                    <property name="maxValue">
+                        <number>8784</number>
+                    </property>
+                    <property name="minValue">
+                        <number>1</number>
+                    </property>
+                    <property name="toolTip" stdset="0">
+                        <string>Number of working hours in a normal year.</string>
+                    </property>
+                </widget>
+                <widget class="KDoubleSpinBox" row="2" column="1">
+                    <property name="name">
+                        <cstring>week</cstring>
+                    </property>
+                    <property name="focusPolicy">
+                        <enum>WheelFocus</enum>
+                    </property>
+                    <property name="maxValue">
+                        <number>168</number>
+                    </property>
+                    <property name="minValue">
+                        <number>1</number>
+                    </property>
+                    <property name="toolTip" stdset="0">
+                        <string>Number of working hours in a normal week.</string>
+                    </property>
+                </widget>
+                <widget class="QLabel" row="0" column="0">
+                    <property name="name">
+                        <cstring>textLabel1</cstring>
+                    </property>
+                    <property name="text">
+                        <string>Hours per year:</string>
+                    </property>
+                    <property name="buddy" stdset="0">
+                        <cstring>year</cstring>
+                    </property>
+                </widget>
+            </grid>
         </widget>
-        <widget class="QLabel" row="1" column="0">
+        <widget class="QGroupBox">
             <property name="name">
-                <cstring>textLabel2</cstring>
+                <cstring>groupBox1</cstring>
             </property>
-            <property name="text">
-                <string>Hours per month:</string>
+            <property name="title">
+                <string>Workinghours</string>
             </property>
-            <property name="buddy" stdset="0">
-                <cstring>month</cstring>
-            </property>
-        </widget>
-        <widget class="QLabel" row="3" column="0">
-            <property name="name">
-                <cstring>textLabel4</cstring>
-            </property>
-            <property name="text">
-                <string>Hours per day:</string>
-            </property>
-            <property name="buddy" stdset="0">
-                <cstring>day</cstring>
-            </property>
-        </widget>
-        <widget class="QLabel" row="2" column="0">
-            <property name="name">
-                <cstring>textLabe3</cstring>
-            </property>
-            <property name="text">
-                <string>Hours per week:</string>
-            </property>
-            <property name="buddy" stdset="0">
-                <cstring>week</cstring>
-            </property>
-        </widget>
-        <widget class="KDoubleSpinBox" row="0" column="1">
-            <property name="name">
-                <cstring>year</cstring>
-            </property>
-            <property name="focusPolicy">
-                <enum>WheelFocus</enum>
-            </property>
-            <property name="maxValue">
-                <number>8784</number>
-            </property>
-            <property name="minValue">
-                <number>1</number>
-            </property>
             <property name="toolTip" stdset="0">
-                <string>Number of working hours in a normal year.</string>
+                <string>Define standard weekly working hours.</string>
             </property>
-        </widget>
-        <widget class="KDoubleSpinBox" row="1" column="1">
-            <property name="name">
-                <cstring>month</cstring>
+            <property name="whatsThis" stdset="0">
+                <string>The working hours defined here will be used
+when there is no calendar defined for a resource.</string>
             </property>
-            <property name="focusPolicy">
-                <enum>WheelFocus</enum>
-            </property>
-            <property name="maxValue">
-                <number>744</number>
-            </property>
-            <property name="minValue">
-                <number>1</number>
-            </property>
-            <property name="toolTip" stdset="0">
-                <string>Number of working hours in a normal month.</string>
-            </property>
+            <hbox>
+                <property name="name">
+                    <cstring>unnamed</cstring>
+                </property>
+                <widget class="K3ListView">
+                    <column>
+                        <property name="text">
+                            <string>Weekday</string>
+                        </property>
+                        <property name="clickable">
+                            <bool>true</bool>
+                        </property>
+                        <property name="resizable">
+                            <bool>true</bool>
+                        </property>
+                    </column>
+                    <column>
+                        <property name="text">
+                            <string>Hours</string>
+                        </property>
+                        <property name="clickable">
+                            <bool>true</bool>
+                        </property>
+                        <property name="resizable">
+                            <bool>true</bool>
+                        </property>
+                    </column>
+                    <property name="name">
+                        <cstring>weekdayList</cstring>
+                    </property>
+                    <property name="sizePolicy">
+                        <sizepolicy>
+                            <hsizetype>7</hsizetype>
+                            <vsizetype>5</vsizetype>
+                            <horstretch>0</horstretch>
+                            <verstretch>0</verstretch>
+                        </sizepolicy>
+                    </property>
+                    <property name="minimumSize">
+                        <size>
+                            <width>160</width>
+                            <height>210</height>
+                        </size>
+                    </property>
+                    <property name="selectionMode" stdset="0">
+                        <enum>Extended</enum>
+                    </property>
+                    <property name="itemMargin">
+                        <number>4</number>
+                    </property>
+                </widget>
+                <widget class="QGroupBox">
+                    <property name="name">
+                        <cstring>editBox</cstring>
+                    </property>
+                    <property name="title">
+                        <string></string>
+                    </property>
+                    <grid>
+                        <property name="name">
+                            <cstring>unnamed</cstring>
+                        </property>
+                        <widget class="KComboBox" row="0" column="0">
+                            <item>
+                                <property name="text">
+                                    <string>Non-working</string>
+                                </property>
+                            </item>
+                            <item>
+                                <property name="text">
+                                    <string>Working</string>
+                                </property>
+                            </item>
+                            <property name="name">
+                                <cstring>state</cstring>
+                            </property>
+                        </widget>
+                        <widget class="QPushButton" row="0" column="1">
+                            <property name="name">
+                                <cstring>bApply</cstring>
+                            </property>
+                            <property name="text">
+                                <string>Apply</string>
+                            </property>
+                        </widget>
+                        <widget class="QGroupBox" row="1" column="0" rowspan="1" colspan="2">
+                            <property name="name">
+                                <cstring>intervalBox</cstring>
+                            </property>
+                            <property name="frameShape">
+                                <enum>GroupBoxPanel</enum>
+                            </property>
+                            <property name="frameShadow">
+                                <enum>Sunken</enum>
+                            </property>
+                            <property name="lineWidth">
+                                <number>1</number>
+                            </property>
+                            <property name="title">
+                                <string></string>
+                            </property>
+                        </widget>
+                    </grid>
+                </widget>
+            </hbox>
         </widget>
-        <widget class="KDoubleSpinBox" row="2" column="1">
-            <property name="name">
-                <cstring>week</cstring>
-            </property>
-            <property name="focusPolicy">
-                <enum>WheelFocus</enum>
-            </property>
-            <property name="maxValue">
-                <number>168</number>
-            </property>
-            <property name="minValue">
-                <number>1</number>
-            </property>
-            <property name="toolTip" stdset="0">
-                <string>Number of working hours in a normal week.</string>
-            </property>
-        </widget>
-        <widget class="KDoubleSpinBox" row="3" column="1">
-            <property name="name">
-                <cstring>day</cstring>
-            </property>
-            <property name="focusPolicy">
-                <enum>WheelFocus</enum>
-            </property>
-            <property name="maxValue">
-                <number>24</number>
-            </property>
-            <property name="minValue">
-                <number>1</number>
-            </property>
-            <property name="toolTip" stdset="0">
-                <string>Number of working hours in a normal day.</string>
-            </property>
-        </widget>
-    </grid>
+    </vbox>
 </widget>
 <customwidgets>
 </customwidgets>
+<tabstops>
+    <tabstop>day</tabstop>
+    <tabstop>weekdayList</tabstop>
+    <tabstop>state</tabstop>
+    <tabstop>bApply</tabstop>
+    <tabstop>year</tabstop>
+    <tabstop>month</tabstop>
+    <tabstop>week</tabstop>
+</tabstops>
 <layoutdefaults spacing="6" margin="11"/>
 <includehints>
     <includehint>knuminput.h</includehint>
-    <includehint>knuminput.h</includehint>
-    <includehint>knuminput.h</includehint>
-    <includehint>knuminput.h</includehint>
+    <includehint>klistview.h</includehint>
+    <includehint>kcombobox.h</includehint>
 </includehints>
 </UI>
Index: kptproject.cc
===================================================================
--- kptproject.cc	(revision 528102)
+++ kptproject.cc	(working copy)
@@ -1,6 +1,6 @@
 /* This file is part of the KDE project
    Copyright (C) 2001 Thomas zander <zander@kde.org>
-   Copyright (C) 2004, 2005 Dag Andersen <danders@get2net.dk>
+   Copyright (C) 2004 - 2006 Dag Andersen <danders@get2net.dk>
 
    This library is free software; you can redistribute it and/or
    modify it under the terms of the GNU Library General Public
@@ -45,7 +45,6 @@
 Project::Project(Node *parent)
     : Node(parent),
       m_accounts(*this),
-      m_defaultCalendar(0),
       m_baselined(false) {
     //kDebug()<<k_funcinfo<<"("<<this<<")"<<endl;
     m_constraint = Node::MustStartOn;
@@ -926,10 +925,6 @@
     }
 }
 
-void Project::setDefaultCalendar(Calendar *cal) {
-    m_defaultCalendar = cal;
-}
-
 bool Project::legalToLink(Node *par, Node *child) {
     //kDebug()<<k_funcinfo<<par.name()<<" ("<<par.numDependParentNodes()<<" parents) "<<child.name()<<" ("<<child.numDependChildNodes()<<" children)"<<endl;
     
Index: kptintervaledit.cc
===================================================================
--- kptintervaledit.cc	(revision 528102)
+++ kptintervaledit.cc	(working copy)
@@ -1,5 +1,5 @@
 /* This file is part of the KDE project
-   Copyright (C) 2004 Dag Andersen <danders@get2net.dk>
+   Copyright (C) 2004 - 2006 Dag Andersen <danders@get2net.dk>
 
    This library is free software; you can redistribute it and/or
    modify it under the terms of the GNU Library General Public
@@ -22,11 +22,14 @@
 
 #include <QPushButton>
 #include <QComboBox>
+#include <qheader.h>
 #include <QLabel>
 #include <QLineEdit>
 #include <qdatetimeedit.h>
 #include <qdatetime.h>
 #include <qlistview.h>
+#include <qpair.h>
+#include <qdatetime.h>
 
 #include <klocale.h>
 #include <kdebug.h>
@@ -34,32 +37,19 @@
 namespace KPlato
 {
 
-IntervalEdit::IntervalEdit(const QPtrList<QPair<QTime, QTime> > &intervals, QWidget *parent, const char *name)
-    : KDialogBase( Swallow, i18n("Edit Interval"), Ok|Cancel, Ok, parent, name, true, true)
+IntervalEdit::IntervalEdit(QWidget *parent, const char *name)
+    : IntervalEditImpl(parent)
 {
     //kDebug()<<k_funcinfo<<endl;
-    dia = new IntervalEditImpl(intervals, this);
 
-    setMainWidget(dia);
-    enableButtonOK(false);
-
-    connect(dia, SIGNAL(obligatedFieldsFilled(bool) ), SLOT(enableButtonOK(bool)));
-    connect(dia, SIGNAL(enableButtonOk(bool)), SLOT(enableButtonOK(bool)));
 }
 
-QPtrList<QPair<QTime, QTime> > IntervalEdit::intervals() const {
-    return dia->intervals();
-}
-
-
-IntervalEditImpl::IntervalEditImpl(const QPtrList<QPair<QTime, QTime> > &intervals, QWidget *parent)
+//--------------------------------------------
+IntervalEditImpl::IntervalEditImpl(QWidget *parent)
     : IntervalEditBase(parent) {
 
+    intervalList->header()->setStretchEnabled(true);
     intervalList->setSortColumn(0);
-    QPtrListIterator<QPair<QTime, QTime> > it = intervals;
-    for (; it.current(); ++it) {
-        new IntervalItem(intervalList, it.current()->first, it.current()->second);
-    }
 
     connect(bClear, SIGNAL(clicked()), SLOT(slotClearClicked()));
     connect(bAddInterval, SIGNAL(clicked()), SLOT(slotAddIntervalClicked()));
@@ -67,22 +57,16 @@
 
 }
 
-
-void IntervalEditImpl::slotEnableButtonOk(bool on) {
-    emit enableButtonOk(on);
-}
-
-void IntervalEditImpl::slotCheckAllFieldsFilled() {
-    emit obligatedFieldsFilled(true); //FIXME
-}
-
 void IntervalEditImpl::slotClearClicked() {
+    bool c = intervalList->firstChild() != 0;
     intervalList->clear();
+    if (c)
+        emit changed();
 }
 
 void IntervalEditImpl::slotAddIntervalClicked() {
     new IntervalItem(intervalList, startTime->time(), endTime->time());
-    slotEnableButtonOk(true);
+    emit changed();
 }
 
 void IntervalEditImpl::slotIntervalSelectionChanged(QListViewItem *item) {
@@ -104,6 +88,14 @@
     return l;
 }
 
+void IntervalEditImpl::setIntervals(const QPtrList<QPair<QTime, QTime> > &intervals) const {
+    intervalList->clear();
+    QPtrListIterator<QPair<QTime, QTime> > it =intervals;
+    for (; it.current(); ++it) {
+        new IntervalItem(intervalList, it.current()->first, it.current()->second);
+    }
+}
+
 }  //KPlato namespace
 
 #include "kptintervaledit.moc"
Index: kptstandardworktimedialog.h
===================================================================
--- kptstandardworktimedialog.h	(revision 528102)
+++ kptstandardworktimedialog.h	(working copy)
@@ -1,5 +1,5 @@
 /* This file is part of the KDE project
-   Copyright (C) 2004 Dag Andersen <danders@get2net.dk>
+   Copyright (C) 2004 - 2006 Dag Andersen <danders@get2net.dk>
 
    This library is free software; you can redistribute it and/or
    modify it under the terms of the GNU Library General Public
@@ -34,6 +34,7 @@
 
 class Project;
 class Part;
+class IntervalEditImpl;
 
 class StandardWorktimeDialogImpl : public StandardWorktimeDialogBase {
     Q_OBJECT
@@ -54,7 +55,11 @@
     void slotMonthChanged(double);
     void slotWeekChanged(double);
     void slotDayChanged(double);
-    
+    void slotIntervalChanged();
+    void slotApplyClicked();
+    void slotEnableButtonApply(bool);
+    void slotWeekdaySelected();
+    void slotStateChanged(int);
 signals:
     void obligatedFieldsFilled(bool yes);
     void enableButtonOk(bool);
@@ -65,6 +70,7 @@
     double m_month;
     double m_week;
     double m_day;
+    IntervalEditImpl *m_intervalEdit;
 };
 
 class StandardWorktimeDialog : public KDialogBase {
Index: kptcalendar.h
===================================================================
--- kptcalendar.h	(revision 528102)
+++ kptcalendar.h	(working copy)
@@ -1,5 +1,5 @@
 /* This file is part of the KDE project
-   Copyright (C) 2003 - 2005 Dag Andersen <danders@get2net.dk>
+   Copyright (C) 2003 - 2006 Dag Andersen <danders@get2net.dk>
 
    This library is free software; you can redistribute it and/or
    modify it under the terms of the GNU Library General Public
@@ -376,6 +376,8 @@
     bool load(QDomElement &element);
     void save(QDomElement &element) const;
 
+    Calendar *calendar() const { return m_calendar; }
+    
 protected:
     void init();
     
@@ -385,6 +387,8 @@
     Duration m_week;
     Duration m_day;
     
+    Calendar *m_calendar;
+    
 #ifndef NDEBUG
 public:
     void printDebug(QCString indent="");
Index: kptintervaledit.h
===================================================================
--- kptintervaledit.h	(revision 528102)
+++ kptintervaledit.h	(working copy)
@@ -1,5 +1,5 @@
 /* This file is part of the KDE project
-   Copyright (C) 2004 Dag Andersen <danders@get2net.dk>
+   Copyright (C) 2004 - 2006 Dag Andersen <danders@get2net.dk>
 
    This library is free software; you can redistribute it and/or
    modify it under the terms of the GNU Library General Public
@@ -27,6 +27,7 @@
 #include <QString>
 #include <qptrlist.h>
 #include <qpair.h>
+#include <QWidget>
 
 namespace KPlato
 {
@@ -34,32 +35,25 @@
 class IntervalEditImpl : public IntervalEditBase {
     Q_OBJECT
 public:
-    IntervalEditImpl(const QPtrList<QPair<QTime, QTime> > &intervals, QWidget *parent);
+    IntervalEditImpl(QWidget *parent);
     
     QPtrList<QPair<QTime, QTime> > intervals() const;
+    void setIntervals(const QPtrList<QPair<QTime, QTime> > &intervals) const;
     
 private slots:
-    void slotCheckAllFieldsFilled();
-    void slotEnableButtonOk(bool on);
-    
     void slotClearClicked();
     void slotAddIntervalClicked();
     void slotIntervalSelectionChanged(QListViewItem *item);
 signals:
-    void obligatedFieldsFilled(bool yes);
-    void enableButtonOk(bool);
+    void changed();
     
 };
 
-class IntervalEdit : public KDialogBase {
+class IntervalEdit : public IntervalEditImpl {
     Q_OBJECT
 public:
-    IntervalEdit(const QPtrList<QPair<QTime, QTime> > &intervals, QWidget *parent=0, const char *name=0);
+    IntervalEdit(QWidget *parent=0, const char *name=0);
     
-    QPtrList<QPair<QTime, QTime> > intervals() const;
-    
-private:
-    IntervalEditImpl *dia;
 };
 
 }  //KPlato namespace
Index: kptpart.cc
===================================================================
--- kptpart.cc	(revision 528102)
+++ kptpart.cc	(working copy)
@@ -2,6 +2,7 @@
    Copyright (C) 1998, 1999, 2000 Torben Weis <weis@kde.org>
    Copyright (C) 2004, 2005 Dag Andersen <danders@get2net.dk>
    Copyright (C) 2006 Raphael Langerhorst <raphael.langerhorst@kdemail.net>
+   Copyright (C) 2006 Dag Andersen <danders@get2net.dk>
 
    This library is free software; you can redistribute it and/or
    modify it under the terms of the GNU Library General Public
@@ -42,7 +43,7 @@
 #include <KoCommandHistory.h>
 #include <KoGlobal.h>
 
-#define CURRENT_SYNTAX_VERSION "0.3"
+#define CURRENT_SYNTAX_VERSION "0.4"
 
 namespace KPlato
 {
Index: kptintervaleditbase.ui
===================================================================
--- kptintervaleditbase.ui	(revision 528102)
+++ kptintervaleditbase.ui	(working copy)
@@ -10,7 +10,7 @@
             <x>0</x>
             <y>0</y>
             <width>278</width>
-            <height>248</height>
+            <height>237</height>
         </rect>
     </property>
     <property name="caption">
@@ -41,6 +41,9 @@
             <property name="focusPolicy">
                 <enum>NoFocus</enum>
             </property>
+            <property name="showSortIndicator">
+                <bool>true</bool>
+            </property>
             <property name="resizeMode">
                 <enum>AllColumns</enum>
             </property>
@@ -53,51 +56,15 @@
                 <property name="name">
                     <cstring>unnamed</cstring>
                 </property>
-                <widget class="QLayoutWidget">
+                <widget class="QTimeEdit">
                     <property name="name">
-                        <cstring>layout4</cstring>
+                        <cstring>startTime</cstring>
                     </property>
-                    <hbox>
-                        <property name="name">
-                            <cstring>unnamed</cstring>
-                        </property>
-                        <widget class="QLabel">
-                            <property name="name">
-                                <cstring>textLabel1</cstring>
-                            </property>
-                            <property name="text">
-                                <string>From:</string>
-                            </property>
-                        </widget>
-                        <widget class="QTimeEdit">
-                            <property name="name">
-                                <cstring>startTime</cstring>
-                            </property>
-                        </widget>
-                    </hbox>
                 </widget>
-                <widget class="QLayoutWidget">
+                <widget class="QTimeEdit">
                     <property name="name">
-                        <cstring>layout5</cstring>
+                        <cstring>endTime</cstring>
                     </property>
-                    <hbox>
-                        <property name="name">
-                            <cstring>unnamed</cstring>
-                        </property>
-                        <widget class="QLabel">
-                            <property name="name">
-                                <cstring>textLabel2</cstring>
-                            </property>
-                            <property name="text">
-                                <string>To:</string>
-                            </property>
-                        </widget>
-                        <widget class="QTimeEdit">
-                            <property name="name">
-                                <cstring>endTime</cstring>
-                            </property>
-                        </widget>
-                    </hbox>
                 </widget>
             </hbox>
         </widget>
Index: templates/Simple/8HourDay-40HourWeek.kplatot
===================================================================
--- templates/Simple/8HourDay-40HourWeek.kplatot	(revision 528102)
+++ templates/Simple/8HourDay-40HourWeek.kplatot	(working copy)
@@ -1,27 +1,27 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<!DOCTYPE DOC>
-<DOC mime="application/x-vnd.kde.kplato" syntaxVersion="0" >
-
- <project id="0" scheduling="MustStartOn" >
-  <calendar id="1" name="Base" >
-   <weekday day="0" state="2" >
-    <interval end="16:00:00" start="08:00:00" />
-   </weekday>
-   <weekday day="1" state="2" >
-    <interval end="16:00:00" start="08:00:00" />
-   </weekday>
-   <weekday day="2" state="2" >
-    <interval end="16:00:00" start="08:00:00" />
-   </weekday>
-   <weekday day="3" state="2" >
-    <interval end="16:00:00" start="08:00:00" />
-   </weekday>
-   <weekday day="4" state="2" >
-    <interval end="16:00:00" start="08:00:00" />
-   </weekday>
-   <weekday day="5" state="1" />
-   <weekday day="6" state="1" />
-  </calendar>
-  <standard-worktime  day="8h0m" week="40h0m" month="176h0m" year="1760h0m" />
+<!DOCTYPE kplato>
+<kplato mime="application/x-vnd.kde.kplato" version="0.4" editor="KPlato" >
+ <project scheduling="MustStartOn" >
+  <standard-worktime day="8h0m" week="40h0m" month="176h0m" year="1760h0m" >
+   <calendar id="" name="Base" >
+    <weekday day="0" state="2" >
+     <interval end="16:00:00" start="08:00:00" />
+    </weekday>
+    <weekday day="1" state="2" >
+     <interval end="16:00:00" start="08:00:00" />
+    </weekday>
+    <weekday day="2" state="2" >
+     <interval end="16:00:00" start="08:00:00" />
+    </weekday>
+    <weekday day="3" state="2" >
+     <interval end="16:00:00" start="08:00:00" />
+    </weekday>
+    <weekday day="4" state="2" >
+     <interval end="16:00:00" start="08:00:00" />
+    </weekday>
+    <weekday day="5" state="1" />
+    <weekday day="6" state="1" />
+   </calendar>
+  </standard-worktime>
  </project>
-</DOC>
+</kplato>
