--- koffice/kivio/plugins/kivioconnectortool/Makefile.am	2002/08/30 10:01:35	1.11
+++ koffice/kivio/plugins/kivioconnectortool/Makefile.am	2002/09/08 00:33:32
@@ -3,7 +3,7 @@
 #### WARNING! All changes made in this part will be lost! ####
 ##############################################################
 SUBDIRS = straight_connector 
-lib_LTLIBRARIES =libkivioconnectortool.la
+kde_module_LTLIBRARIES =libkivioconnectortool.la
 
 
 INCLUDES = -I$(top_srcdir)/kivio/kiviopart\
--- koffice/kivio/plugins/kivioselecttool/Makefile.am	2002/08/30 10:01:35	1.9
+++ koffice/kivio/plugins/kivioselecttool/Makefile.am	2002/09/08 00:33:32
@@ -3,7 +3,7 @@
 #### WARNING! All changes made in this part will be lost! ####
 ##############################################################
 SUBDIRS = select_pics 
-lib_LTLIBRARIES =libkivioselecttool.la
+kde_module_LTLIBRARIES =libkivioselecttool.la
 
 
 INCLUDES = -I$(top_srcdir)/kivio/kiviopart\
--- koffice/kivio/plugins/kiviotexttool/Makefile.am	2002/08/30 10:01:35	1.9
+++ koffice/kivio/plugins/kiviotexttool/Makefile.am	2002/09/08 00:33:32
@@ -3,7 +3,7 @@
 #### WARNING! All changes made in this part will be lost! ####
 ##############################################################
 
-lib_LTLIBRARIES =libkiviotexttool.la
+kde_module_LTLIBRARIES =libkiviotexttool.la
 
 
 INCLUDES = -I$(top_srcdir)/kivio/kiviopart\
--- koffice/kivio/plugins/kiviozoomtool/Makefile.am	2002/08/30 10:01:35	1.10
+++ koffice/kivio/plugins/kiviozoomtool/Makefile.am	2002/09/08 00:33:32
@@ -3,7 +3,7 @@
 #### WARNING! All changes made in this part will be lost! ####
 ##############################################################
 SUBDIRS = zoom_pics 
-lib_LTLIBRARIES =libkiviozoomtool.la
+kde_module_LTLIBRARIES =libkiviozoomtool.la
 
 INCLUDES = -I$(top_srcdir)/kivio/kiviopart\
 	-I$(top_srcdir)/kivio/kiviopart/kiviosdk\
