--- koffice/karbon/Makefile.am	2002/08/05 19:52:14	1.44
+++ koffice/karbon/Makefile.am	2002/09/08 00:33:29
@@ -42,7 +42,7 @@ libkarbonpart_la_LIBADD = $(LIB_KOFFICEU
 libkarbonpart_la_METASOURCES = AUTO
 
 karbon_la_SOURCES = main.cc
-karbon_la_LDFLAGS = $(all_libraries) -module -avoid-version
+karbon_la_LDFLAGS = $(all_libraries) -module
 karbon_la_LIBADD = libkarbonpart.la
 
 karbon_SOURCES = dummy.cc
--- koffice/kchart/Makefile.am	2002/07/15 06:25:07	1.98
+++ koffice/kchart/Makefile.am	2002/09/08 00:33:30
@@ -41,7 +41,7 @@ libkchartpart_la_LIBADD = $(top_builddir
 ## The kdeinit loadable module
 lib_LTLIBRARIES = kchart.la
 kchart_la_SOURCES = main.cc
-kchart_la_LDFLAGS = $(all_libraries) -avoid-version -module
+kchart_la_LDFLAGS = $(all_libraries) -module
 kchart_la_LIBADD = $(LIB_KOFFICECORE)
 
 ## The executable
--- koffice/kformula/Makefile.am	2002/07/14 12:25:12	1.70
+++ koffice/kformula/Makefile.am	2002/09/08 00:33:30
@@ -14,7 +14,7 @@ libkformulapart_la_METASOURCES = AUTO
 ## The kdeinit loadable module
 lib_LTLIBRARIES = kformulamain.la
 kformulamain_la_SOURCES = main.cc
-kformulamain_la_LDFLAGS = $(all_libraries) -module -avoid-version
+kformulamain_la_LDFLAGS = $(all_libraries) -module
 kformulamain_la_LIBADD = $(LIB_KOFFICECORE)
 
 ## The executable
--- koffice/kivio/kiviopart/Makefile.am	2002/07/14 12:31:28	1.24
+++ koffice/kivio/kiviopart/Makefile.am	2002/09/08 00:33:31
@@ -141,7 +141,7 @@ EXTRA_DIST = \
 #2########################################################!@#$
 lib_LTLIBRARIES = kivio.la
 kivio_la_SOURCES = main.cpp
-kivio_la_LDFLAGS = $(all_libraries) -module -avoid-version
+kivio_la_LDFLAGS = $(all_libraries) -module
 kivio_la_LIBADD = ../../lib/kofficecore/libkofficecore.la
 
 libkiviopart_la_LDFLAGS = -avoid-version -module $(all_libraries)
--- koffice/kontour/Makefile.am	2002/07/27 10:56:20	1.72
+++ koffice/kontour/Makefile.am	2002/09/08 00:33:33
@@ -14,7 +14,7 @@ rc_DATA = kontour.rc kontour_readonly.rc
 ## The kdeinit loadable module
 lib_LTLIBRARIES = kontour.la
 kontour_la_SOURCES = main.cc
-kontour_la_LDFLAGS = $(all_libraries) -avoid-version -module
+kontour_la_LDFLAGS = $(all_libraries) -module
 kontour_la_LIBADD = ../lib/kofficecore/libkofficecore.la
 
 ## The executable
--- koffice/koshell/Makefile.am	2001/03/22 14:02:17	1.25
+++ koffice/koshell/Makefile.am	2002/09/08 00:33:40
@@ -4,7 +4,7 @@ INCLUDES  = $(KOFFICE_INCLUDES) $(all_in
 ## The kdeinit loadable module
 lib_LTLIBRARIES = koshell.la
 koshell_la_SOURCES = koshell_main.cc koshell_shell.cc
-koshell_la_LDFLAGS = $(all_libraries) -avoid-version -module
+koshell_la_LDFLAGS = $(all_libraries) -module
 koshell_la_LIBADD = $(LIB_KOFFICEUI)
 
 ## The executable
--- koffice/kpresenter/Makefile.am	2002/07/14 12:31:37	1.200
+++ koffice/kpresenter/Makefile.am	2002/09/08 00:33:40
@@ -55,7 +55,7 @@ libkpresenterpart_la_METASOURCES = AUTO
 ## The kdeinit loadable module
 lib_LTLIBRARIES = kpresenter.la
 kpresenter_la_SOURCES = main.cc
-kpresenter_la_LDFLAGS = $(all_libraries) -avoid-version -module
+kpresenter_la_LDFLAGS = $(all_libraries) -module
 kpresenter_la_LIBADD = $(LIB_KOFFICECORE)
 
 ## The executable
--- koffice/kspread/Makefile.am	2002/07/14 12:21:47	1.175
+++ koffice/kspread/Makefile.am	2002/09/08 00:33:40
@@ -52,7 +52,7 @@ libkspreadpart_la_METASOURCES = AUTO
 ## The kdeinit loadable module
 lib_LTLIBRARIES =  kspread.la
 kspread_la_SOURCES = main.cc
-kspread_la_LDFLAGS = $(all_libraries) -avoid-version -module
+kspread_la_LDFLAGS = $(all_libraries) -module
 kspread_la_LIBADD = $(LIB_KOFFICECORE)
 
 ## The executable
--- koffice/kugar/part/Makefile.am	2002/02/27 13:41:41	1.5
+++ koffice/kugar/part/Makefile.am	2002/09/08 00:33:40
@@ -2,7 +2,7 @@ INCLUDES = -I$(srcdir)/../lib $(KOFFICE_
 
 lib_LTLIBRARIES = libkugarpart.la
 libkugarpart_la_SOURCES = kugar_part.cpp
-libkugarpart_la_LDFLAGS = $(all_libraries) -module -avoid-version
+libkugarpart_la_LDFLAGS = $(all_libraries) -module
 libkugarpart_la_LIBADD = ../lib/libkugar.la $(LIB_KPARTS) $(LIB_KDEPRINT)
 
 libkugarpart_la_METASOURCES = AUTO
--- koffice/kword/Makefile.am	2002/07/14 12:01:51	1.188
+++ koffice/kword/Makefile.am	2002/09/08 00:33:40
@@ -57,7 +57,7 @@ kwmailmerge_servicetypedir=$(kde_service
 
 ## The kdeinit loadable module
 kword_la_SOURCES = main.cc
-kword_la_LDFLAGS = $(all_libraries) -avoid-version -module
+kword_la_LDFLAGS = $(all_libraries) -module
 kword_la_LIBADD = ../lib/kofficecore/libkofficecore.la
 
 ## The executable
