--- koffice/lib/kformula/contextstyle.cc	31 May 2003 14:03:15 -0000	1.50
+++ koffice/lib/kformula/contextstyle.cc	9 May 2004 02:58:07 -0000
@@ -118,7 +118,7 @@ void ContextStyle::readConfig( KConfig* 
     QString baseSize = config->readEntry( "baseSize", "20" );
     m_baseSize = baseSize.toInt();
 
-    m_fontStyleName = config->readEntry( "fontStyle", "symbol" );
+    m_fontStyleName = config->readEntry( "fontStyle", "tex" );
 
 #if 0
     m_requestedFonts = config->readListEntry( "usedMathFonts" );
