--- koffice-1.1-beta2.orig/kchart/Makefile.am
+++ koffice-1.1-beta2/kchart/Makefile.am
@@ -49,7 +49,7 @@
 ## The kdeinit loadable module
 lib_LTLIBRARIES = kchart.la
 kchart_la_SOURCES = main.cc
-kchart_la_LDFLAGS = $(all_libraries) -avoid-version -module
+kchart_la_LDFLAGS = $(all_libraries) -module
 kchart_la_LIBADD = ../lib/kofficecore/libkofficecore.la
 
 ## The executable
--- koffice-1.1-beta2.orig/kformula/Makefile.am
+++ koffice-1.1-beta2/kformula/Makefile.am
@@ -14,7 +14,7 @@
 ## The kdeinit loadable module
 lib_LTLIBRARIES = kformulamain.la
 kformulamain_la_SOURCES = main.cc
-kformulamain_la_LDFLAGS = $(all_libraries) -module -avoid-version
+kformulamain_la_LDFLAGS = $(all_libraries) -module
 kformulamain_la_LIBADD = ../lib/kofficecore/libkofficecore.la
 
 ## The executable
--- koffice-1.1-beta2.orig/killustrator/Makefile.am
+++ koffice-1.1-beta2/killustrator/Makefile.am
@@ -20,7 +20,7 @@
 ## The kdeinit loadable module
 lib_LTLIBRARIES = killustrator.la
 killustrator_la_SOURCES = main.cc
-killustrator_la_LDFLAGS = $(all_libraries) -avoid-version -module
+killustrator_la_LDFLAGS = $(all_libraries) -module
 killustrator_la_LIBADD = ../lib/kofficecore/libkofficecore.la
 
 ## The executable
--- koffice-1.1-beta2.orig/kimageshop/Makefile.am
+++ koffice-1.1-beta2/kimageshop/Makefile.am
@@ -12,7 +12,7 @@
 ## The kdeinit loadable module
 lib_LTLIBRARIES = krayon.la
 krayon_la_SOURCES = main.cc
-krayon_la_LDFLAGS = $(all_libraries) -avoid-version -module
+krayon_la_LDFLAGS = $(all_libraries) -module
 krayon_la_LIBADD = ../lib/kofficecore/libkofficecore.la
 
 ## The executable
--- koffice-1.1-beta2.orig/kimageshop/plugins/example/Makefile.am
+++ koffice-1.1-beta2/kimageshop/plugins/example/Makefile.am
@@ -1,6 +1,6 @@
 INCLUDES  = -I$(srcdir)/../core -I$(srcdir)/../ui -I$(srcdir)/../tools $(KOFFICE_INCLUDES) -I$(interfacedir) $(all_includes)
 
-libkisp_example_la_LDFLAGS = $(all_libraries) -avoid-version -module
+libkisp_example_la_LDFLAGS = $(all_libraries) -module
 #libkisp_example_la_LIBADD =  ../core/libkiscore.la ../ui/libkisui.la ../tools/libkistools.la \
 #	../../lib/kofficeui/libkofficeui.la
 
--- koffice-1.1-beta2.orig/kivio/kiviopart/Makefile.am
+++ koffice-1.1-beta2/kivio/kiviopart/Makefile.am
@@ -131,7 +131,7 @@
 #2########################################################!@#$
 lib_LTLIBRARIES = kivio.la
 kivio_la_SOURCES = main.cpp
-kivio_la_LDFLAGS = $(all_libraries) -module -avoid-version
+kivio_la_LDFLAGS = $(all_libraries) -module
 kivio_la_LIBADD = ../../lib/kofficecore/libkofficecore.la
 
 libkiviopart_la_LDFLAGS = -avoid-version -module $(LIB_KOFFICEUI) $(all_libraries)
--- koffice-1.1-beta2.orig/kivio/plugins/kivioselecttool/Makefile.am
+++ koffice-1.1-beta2/kivio/plugins/kivioselecttool/Makefile.am
@@ -27,7 +27,7 @@
 ##############################################################
 ####                  END KDEStudio part                  ####
 #2########################################################!@#$
-libkivioselecttool_la_LDFLAGS = -avoid-version -module -no-undefined
+libkivioselecttool_la_LDFLAGS = -module -no-undefined
 libkivioselecttool_la_LIBADD = $(all_libraries) $(top_builddir)/kivio/kiviopart/libkiviopart.la
 
 rc_DATA = kivioselecttool.rc
--- koffice-1.1-beta2.orig/kivio/plugins/kivioconnectortool/Makefile.am
+++ koffice-1.1-beta2/kivio/plugins/kivioconnectortool/Makefile.am
@@ -31,7 +31,7 @@
 ##############################################################
 ####                  END KDEStudio part                  ####
 #2########################################################!@#$
-libkivioconnectortool_la_LDFLAGS = -avoid-version -module -no-undefined
+libkivioconnectortool_la_LDFLAGS = -module -no-undefined
 libkivioconnectortool_la_LIBADD = $(all_libraries) $(top_builddir)/kivio/kiviopart/libkiviopart.la
 
 rc_DATA = kivioconnectortool.rc
--- koffice-1.1-beta2.orig/kivio/plugins/kiviotexttool/Makefile.am
+++ koffice-1.1-beta2/kivio/plugins/kiviotexttool/Makefile.am
@@ -31,7 +31,7 @@
 ##############################################################
 ####                  END KDEStudio part                  ####
 #2########################################################!@#$
-libkiviotexttool_la_LDFLAGS = -avoid-version -module -no-undefined
+libkiviotexttool_la_LDFLAGS = -module -no-undefined
 libkiviotexttool_la_LIBADD = $(all_libraries) $(top_builddir)/kivio/kiviopart/libkiviopart.la
 
 rc_DATA = kiviotexttool.rc
--- koffice-1.1-beta2.orig/kivio/plugins/kiviozoomtool/Makefile.am
+++ koffice-1.1-beta2/kivio/plugins/kiviozoomtool/Makefile.am
@@ -27,7 +27,7 @@
 ##############################################################
 ####                  END KDEStudio part                  ####
 #2########################################################!@#$
-libkiviozoomtool_la_LDFLAGS = -avoid-version -module -no-undefined
+libkiviozoomtool_la_LDFLAGS = -module -no-undefined
 libkiviozoomtool_la_LIBADD = $(all_libraries) $(top_builddir)/kivio/kiviopart/libkiviopart.la
 
 rc_DATA = kiviozoomtool.rc
--- koffice-1.1-beta2.orig/koshell/Makefile.am
+++ koffice-1.1-beta2/koshell/Makefile.am
@@ -4,7 +4,7 @@
 ## The kdeinit loadable module
 lib_LTLIBRARIES = koshell.la
 koshell_la_SOURCES = koshell_main.cc koshell_shell.cc
-koshell_la_LDFLAGS = $(all_libraries) -avoid-version -module
+koshell_la_LDFLAGS = $(all_libraries) -module
 koshell_la_LIBADD = $(LIB_KOFFICEUI)
 
 ## The executable
--- koffice-1.1-beta2.orig/kpresenter/Makefile.am
+++ koffice-1.1-beta2/kpresenter/Makefile.am
@@ -38,7 +38,7 @@
 ## The kdeinit loadable module
 lib_LTLIBRARIES = kpresenter.la
 kpresenter_la_SOURCES = main.cc
-kpresenter_la_LDFLAGS = $(all_libraries) -avoid-version -module
+kpresenter_la_LDFLAGS = $(all_libraries) -module
 kpresenter_la_LIBADD = ../lib/kofficecore/libkofficecore.la
 
 ## The executable
--- koffice-1.1-beta2.orig/kspread/Makefile.am
+++ koffice-1.1-beta2/kspread/Makefile.am
@@ -38,7 +38,7 @@
 ## The kdeinit loadable module
 lib_LTLIBRARIES =  kspread.la
 kspread_la_SOURCES = main.cc
-kspread_la_LDFLAGS = $(all_libraries) -avoid-version -module
+kspread_la_LDFLAGS = $(all_libraries) -module
 kspread_la_LIBADD = ../lib/kofficecore/libkofficecore.la
 
 ## The executable
--- koffice-1.1-beta2.orig/kugar/part/Makefile.am
+++ koffice-1.1-beta2/kugar/part/Makefile.am
@@ -2,7 +2,7 @@
 
 lib_LTLIBRARIES = libkugarpart.la
 libkugarpart_la_SOURCES = kugar_part.cpp
-libkugarpart_la_LDFLAGS = $(all_libraries) -module -avoid-version
+libkugarpart_la_LDFLAGS = $(all_libraries) -module
 libkugarpart_la_LIBADD = ../lib/libkugar.la $(LIB_KPARTS) $(LIB_KDEPRINT)
 
 libkugarpart_la_METASOURCES = AUTO
--- koffice-1.1-beta2.orig/kugar/part/kugar.desktop
+++ koffice-1.1-beta2/kugar/part/kugar.desktop
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Name=Kugar
 Exec=kugar
-Icon=
+Icon=kugar
 MimeType=application/x-kugar;
 Type=Application
 DocPath=
--- koffice-1.1-beta2.orig/kugar/part/kugar_part.cpp
+++ koffice-1.1-beta2/kugar/part/kugar_part.cpp
@@ -78,7 +78,7 @@
 
 	if (!s_instance)
 	{
-		KAboutData *about = new KAboutData(PACKAGE,
+		KAboutData *about = new KAboutData("kugar",
 						   I18N_NOOP("Kugar"),
 						   VERSION);
 
--- koffice-1.1-beta2.orig/kugar/shell/main.cpp
+++ koffice-1.1-beta2/kugar/shell/main.cpp
@@ -18,7 +18,7 @@
 
 int main(int argc,char **argv)
 {
-	KAboutData about(PACKAGE,I18N_NOOP("Kugar"),VERSION,
+	KAboutData about("kugar",I18N_NOOP("Kugar"),VERSION,
 			 I18N_NOOP("A template driven report viewer for XML data."),
 			 KAboutData::License_GPL_V2,
 			 "Copyright (c) 1999-2000 Mutiny Bay Software\n"
--- koffice-1.1-beta2.orig/kword/Makefile.am
+++ koffice-1.1-beta2/kword/Makefile.am
@@ -24,7 +24,7 @@
 ## The kdeinit loadable module
 lib_LTLIBRARIES = kword.la
 kword_la_SOURCES = main.cc
-kword_la_LDFLAGS = $(all_libraries) -avoid-version -module
+kword_la_LDFLAGS = $(all_libraries) -module
 kword_la_LIBADD = ../lib/kofficecore/libkofficecore.la
 
 ## The executable
--- koffice-1.1-beta2.orig/lib/kwmf/Makefile.am
+++ koffice-1.1-beta2/lib/kwmf/Makefile.am
@@ -2,7 +2,7 @@
 
 INCLUDES= -I$(srcdir) $(all_includes)
 # Why is it a -module ? Shouldn't it have -version-info instead ? (DF)
-libkwmf_la_LDFLAGS	= -module -avoid-version -no-undefined
+libkwmf_la_LDFLAGS	= -module -no-undefined
 libkwmf_la_LIBADD = $(KOFFICE_LIBS)
 
 ####### Files
--- koffice-1.1-beta2.orig/admin/Makefile.common
+++ koffice-1.1-beta2/admin/Makefile.common
@@ -86,22 +86,32 @@
 	pwd=`pwd` ;\
 	dirs=`find . -type d | grep -v CVS | sed -e "s#^./##"` ;\
 	for i in $$dirs; do \
-	  if test ! -d "$$pwd/$$i/CVS"; then \
-	    echo "D $$i" ;\
-	    rm -rf "$$pwd/$$i"; \
-	    continue; \
-	  fi ;\
-	  cd $$pwd/$$i ;\
-	  for f in * .*; do \
-	    if test ! -d "$$f"; then \
-	      if grep "^/$$f/" CVS/Entries > /dev/null; then \
-		a="b"; \
-	      else \
-	        echo "F $$i/$$f"; \
-		rm -f "$$pwd/$$i/$$f"; \
-	      fi; \
-	    fi ; \
-	  done; \
+	  case "$$i" in \
+	    debian) \
+		  echo "Ignoring files in $$i"; \
+		;; \
+		*/HTML) \
+		  echo "Ignoring files in $$i"; \
+		;; \
+		*) \
+	      if test ! -d "$$pwd/$$i/CVS"; then \
+	        echo "D $$i" ;\
+	        rm -rf "$$pwd/$$i"; \
+	        continue; \
+	      fi ;\
+	      cd $$pwd/$$i ;\
+	      for f in * .*; do \
+	        if test ! -d "$$f"; then \
+	          if grep "^/$$f/" CVS/Entries > /dev/null; then \
+		    a="b"; \
+	          else \
+	            echo "F $$i/$$f"; \
+		    rm -f "$$pwd/$$i/$$f"; \
+	          fi; \
+	        fi ; \
+	      done; \
+	    ;; \
+	  esac; \
 	done
 
 
