Index: kexiutils/utils.cpp
===================================================================
--- kexiutils/utils.cpp (Revision 579298)
+++ kexiutils/utils.cpp (Arbeitskopie)
@@ -101,6 +101,8 @@
 QString KexiUtils::fileDialogFilterString(const QString& mimeString, bool kdeFormat)
 {
        KMimeType::Ptr ptr = KMimeType::mimeType(mimeString);
+       if ( ! ptr )
+               ptr = KMimeType::defaultMimeTypePtr();
        return fileDialogFilterString( ptr, kdeFormat );
 }

Index: widget/kexiprjtypeselector.cpp
===================================================================
--- widget/kexiprjtypeselector.cpp      (Revision 579298)
+++ widget/kexiprjtypeselector.cpp      (Arbeitskopie)
@@ -27,11 +27,10 @@
        : KexiPrjTypeSelectorBase( parent, "KexiPrjTypeSelector" )
 {
        QString none;
-       icon_file->setPixmap(
-               KGlobal::iconLoader()->loadIcon( KMimeType::mimeType(
-                       KexiDB::Driver::defaultFileBasedDriverMimeType() )->icon(none,0), KIcon::Desktop, 48
-               )
-       );
+       KMimeType::Ptr mime = KMimeType::mimeType( KexiDB::Driver::defaultFileBasedDriverMimeType() );
+       if ( ! mime )
+           mime = KMimeType::defaultMimeTypePtr();
+       icon_file->setPixmap( KGlobal::iconLoader()->loadIcon( mime->icon(none,0), KIcon::Desktop, 48 ) );
        icon_file->setFixedSize(icon_file->pixmap()->size()/2);
        icon_server->setPixmap( DesktopIcon("network", 48) );
        icon_server->setFixedSize(icon_server->pixmap()->size()/2);
Index: main/startup/KexiStartupDialog.cpp
===================================================================
--- main/startup/KexiStartupDialog.cpp  (Revision 579298)
+++ main/startup/KexiStartupDialog.cpp  (Arbeitskopie)
@@ -129,9 +129,16 @@
        {
                result = 0;
                QString none, iconname;
-               iconname = KMimeType::mimeType( KexiDB::Driver::defaultFileBasedDriverMimeType() )->icon(none,0);
+               KMimeType::Ptr defmime = KMimeType::mimeType( KexiDB::Driver::defaultFileBasedDriverMimeType() );
+               if( ! defmime )
+                   defmime = KMimeType::defaultMimeTypePtr();
+               iconname = defmime->icon(none,0);
                kexi_sqlite_icon = KGlobal::iconLoader()->loadIcon( iconname, KIconLoader::Desktop );
-               iconname = KMimeType::mimeType("application/x-kexiproject-shortcut")->icon(none,0);
+
+               KMimeType::Ptr shortcutmime = KMimeType::mimeType("application/x-kexiproject-shortcut");
+               if( ! shortcutmime )
+                   shortcutmime = KMimeType::defaultMimeTypePtr();
+               iconname = shortcutmime->icon(none,0);
                kexi_shortcut_icon = KGlobal::iconLoader()->loadIcon( iconname, KIconLoader::Desktop );
                prj_selector = 0;
                chkDoNotShow = 0;
Index: main/startup/KexiStartupFileDialog.cpp
===================================================================
--- main/startup/KexiStartupFileDialog.cpp      (Revision 579298)
+++ main/startup/KexiStartupFileDialog.cpp      (Arbeitskopie)
@@ -183,7 +183,8 @@
                        continue;
                filter += KexiUtils::fileDialogFilterString(*it);
                mime = KMimeType::mimeType(*it);
-               allfilters += mime->patterns();
+               if (mime)
+                   allfilters += mime->patterns();
        }

        if (m_excludedMimeTypes.find("all/allfiles")==m_excludedMimeTypes.end())
Index: main/startup/KexiConnSelector.cpp
===================================================================
--- main/startup/KexiConnSelector.cpp   (Revision 579298)
+++ main/startup/KexiConnSelector.cpp   (Arbeitskopie)
@@ -111,7 +111,10 @@
 {
        d->conn_set = &conn_set;
        d->startDirOrVariable = startDirOrVariable;
-       QString none, iconname = KMimeType::mimeType( KexiDB::Driver::defaultFileBasedDriverMimeType() )->icon(none,0);
+       KMimeType::Ptr mime = KMimeType::mimeType( KexiDB::Driver::defaultFileBasedDriverMimeType() );
+       if ( ! mime )
+           mime = KMimeType::defaultMimeTypePtr();
+       QString none, iconname = mime->icon(none,0);
        const QPixmap &icon = KGlobal::iconLoader()->loadIcon( iconname, KIconLoader::Desktop );
        setIcon( icon );

Index: main/startup/KexiNewProjectWizard.cpp
===================================================================
--- main/startup/KexiNewProjectWizard.cpp       (Revision 579298)
+++ main/startup/KexiNewProjectWizard.cpp       (Arbeitskopie)
@@ -106,11 +106,12 @@
        m_prjtype_sel->lv_types->setItemMargin( KDialogBase::marginHint() );
        QString none;
        d->lvi_file = new K3ListViewItem( m_prjtype_sel->lv_types, i18n("New Project Stored in File") );
-       d->lvi_file->setPixmap(0,
-               KGlobal::iconLoader()->loadIcon( KMimeType::mimeType(
-                       KexiDB::Driver::defaultFileBasedDriverMimeType() )->icon(none,0), KIconLoader::Desktop
-               )
-       );
+
+       KMimeType::Ptr mime = KMimeType::mimeType( KexiDB::Driver::defaultFileBasedDriverMimeType() );
+        if ( ! mime )
+           mime = KMimeType::defaultMimeTypePtr();
+       d->lvi_file->setPixmap(0, KGlobal::iconLoader()->loadIcon( mime->icon(none,0), KIconLoader::Desktop ) );
+
        d->lvi_file->setMultiLinesEnabled( true );
        d->lvi_server = new K3ListViewItem( m_prjtype_sel->lv_types, d->lvi_file,
                i18n("New Project Stored on Database Server") );
Index: main/startup/KexiProjectSelector.cpp
===================================================================
--- main/startup/KexiProjectSelector.cpp        (Revision 579298)
+++ main/startup/KexiProjectSelector.cpp        (Arbeitskopie)
@@ -107,7 +107,10 @@
 {
        d->showProjectNameColumn = showProjectNameColumn;
        d->showConnectionColumns = showConnectionColumns;
-       QString none, iconname = KMimeType::mimeType( KexiDB::Driver::defaultFileBasedDriverMimeType() )->icon(none,0);
+       KMimeType::Ptr mime = KMimeType::mimeType( KexiDB::Driver::defaultFileBasedDriverMimeType() );
+       if ( ! mime )
+           mime = KMimeType::defaultMimeTypePtr();
+       QString none, iconname = mime->icon(none,0);
        d->fileicon = KGlobal::iconLoader()->loadIcon( iconname, KIconLoader::Desktop );
        setIcon( d->fileicon );
        d->dbicon = SmallIcon("database");
Index: migration/importwizard.cpp
===================================================================
--- migration/importwizard.cpp  (Revision 579298)
+++ migration/importwizard.cpp  (Arbeitskopie)
@@ -163,7 +163,7 @@
                KMimeType::Ptr mimeTypePtr = KMimeType::mimeType(m_predefinedMimeType);
                msg = i18n("<qt>Database Importing wizard is about to import <nobr>\"%1\"</nobr> file "
                "of type \"%2\" into a Kexi database.</qt>")
-                       .arg(QDir::convertSeparators(m_predefinedDatabaseName)).arg(mimeTypePtr->comment());
+                       .arg(QDir::convertSeparators(m_predefinedDatabaseName)).arg(mimeTypePtr ? mimeTypePtr->comment() : "");
        }
        else {
                msg = i18n("Database Importing wizard allows you to import an existing database "
 
